### PR TITLE
feat: company user creation

### PR DESCRIPTION
**Pull request by AasmundN**
# Changes
- `POST /api/code` is only accessible by administrators and creates a new registration code in the database. The endpoint accepts a body with the key `companyUID`. A registration code is connected to a company and has the following format in the database.
```ts
{
   companyUID: string
   code: string
}
```
- `GET /api/code/:code` validates the code passed as a route parameter. The result body has the following format.
```ts
{
   isValid: boolean
}
```
- `DELETE /api/code` is only accessible by administrators and deletes the code passed through the key `code` in the request body.
- Server utility function `deleteCode` to delete registration codes.
- Server utility function `randomString` to generate a random string with a specified length.
- Updated user endpoints and composables to allow creation of company users, utilising the registration code endpoints. 

# Creating users

The server endpoint for creating users has become quite complex. It can be used both for creating new user and modifying existing users in the database. At some point I will write an extensive guide on how to use the endpoint in the wiki. For now the most important thing is how to create new users.

```ts
// create basic user
registerUser({
   email: 'basic@test.no',
   name: 'Basic User',
   password: 'password',
   studyProgram: 'elsus',
   currentYear: 3,
})

// create company user using the code stored in registrationCode
registerUser({
   email: 'company@test.no',
   name: 'Company User',
   password: 'password',
}, registrationCode)
```

When calling the `registerUser` function  remember to handle errors properly with _trycatch_ blocks. I have created three test users all with the same password `password`: basic@test.no, company@test.no, and admin@test.no.

### CODE REVIEW

**Review by JStoresund — APPROVED** [review]
LGTM. All of the requests work as described in the issue.

File: components/user/SignupForm.vue
Comment (ipeglin) [inline components/user/SignupForm.vue:1]: Remove this file if the base should be set to #92

**Review by ipeglin — APPROVED** [review]
LGTM

**Review by ipeglin — APPROVED** [review]
LGTM

**Pull request merged**